core/src/main/scala/kafka/server/DelayedOperation.scala

```diff
@@ -253,6 +253,9 @@ final class DelayedOperationPurgatory[T <: DelayedOperation](purgatoryName: Stri
    */
   def delayed: Int = timeoutTimer.size
 
+  /**
+    * Cancel watching on any delayed operations for the given key. Note the operation will not be completed
```

Comment:
Since we are turning off reaper for DelayedTxnMarker, could we change tryCompleteElseWatch() to not add the operation to timeoutTimer if the reaper is disabled? Otherwise, we will be leaking memory.